src/playground/service.clj

```diff
@@ -41,6 +57,25 @@
   {:status 200
    :body   {:temperature temperature :orientation orientation}})
 
+;;auth interceptors
+
+(def session-auth-backend
+  (session-backend
+   {:authfn (fn [request]
+              (let [{:keys [username password]} request
+                    known-user                  (get (session.login/all-usernames request) username)]
```

Comment:
`known-user` is unused. In general `all-usernames` seems dubious. Imagine there are 1M users: you don't want to query 1M users (has a cost to both the DB and the server's RAM) on each login